CMakeLists.txt

```diff
@@ -292,6 +292,10 @@ if (NOT USE_SIMD STREQUAL "")
             else ()
                 add_definitions ("-m${feature}")
             endif ()
+            if (feature STREQUAL "fma" AND CMAKE_COMPILER_IS_CLANG)
+                # Clang needs an explicit flag to make fma ops
+                add_definitions ("-ffp-contract=fast")
```

Comment:
On my machine, `gcc -mfma` add the pre-processor symbol `__FMA__=1`. Maybe we could use that to detect when `-mfma` was specified on the command line? (It also has defines for AVX, SSE4.1, POPCNT and others that are implied by `-mfma`).

If we can detect it this way -- maybe we can go the intrinsics route and worry about changing the default of `fp-contract` another day?

As an example of how much leeway the compilers have, you can compare this snippet [1] with clang and gcc under various versions and under various flavors of fp-contract.

[1] https://goo.gl/NOvwhR